### ISSUE TITLE
Rewrite renderer using SVG/canvas

### ISSUE THREAD

**Issue by tjvr** [tjvr/scratchblocks#89]
The Scratch site [recently dropped support for IE 8](https://scratch.mit.edu/info/faq/#requirements), and now only support IE 9 or above.

We can't fix #17 using DOM/CSS; @nathan tried to do this for [Visual](//github.com/nathan/visual), and eventually settled upon a hybrid DOM/canvas solution.

Since we only need static images, we should output either just canvas or just SVG (ideally, whichever is faster). This will:

- allow higher accuracy (eg. fix #17)
- make PNG screenshots easier to generate (eg. for SO posts)
- make PDFs easier to generate (eg. for Code Club projects)
- drop our jQuery dependency (we could do this already, but since the Scratch site requires it anyway I've never bothered!)

## Parser output

The new renderer needs to interpret the AST format output by the parser. You can get an idea of what this looks like by using `parse_scripts`, which outputs in the AST format:

```
> scratchblocks2.parse_scripts("when flag clicked")

[
    [
        {
            "blockid": "when @green-flag clicked",
            "category": "events",
            "image_replacement": "green-flag",
            "lang": "en",
            "pieces": [
                "when ",
                "@",
                " clicked"
            ],
            "shape": "hat"
        }
    ]
]
```

I'd also like to rewrite the parser at some point (see #90), but for now we can assume that the AST format will remain the same. :-)

**Comment by tjvr** [comment]
@as-com You made some progress on this? :)

**Comment by as-com** [comment]
Here's what I have right now: https://jsfiddle.net/ascom/zdfscywv/

There's no layout or proper block sizing, but at least there's pixel-perfect block rendering.

**Comment by tjvr** [comment]
@as-com Thanks so much for making this proof-of-concept!

I've started writing a layout engine, based on the shapes and filters from your code. Here's a screenshot:

![screen shot 2016-03-28 at 15 42 05](https://cloud.githubusercontent.com/assets/639289/14080242/a7f173a2-f4fb-11e5-8027-e9c54869f142.png)

You can even [export the SVG](https://dl.dropboxusercontent.com/u/9598124/Screenshots/foo.svg), although compatibility is iffy (eg. Inkscape works fine, Sketch doesn't understand the filters).

:D

**Comment by liam4** [comment]
Shiny! Also.. really nice boolean corners.

**Comment by tjvr** [comment]
@liam4 Thanks! Fixing long-standing bugs like #17 is a large part of the motivation for this :P

**Comment by tjvr** [comment]
Fixed via #94. :D
